### PR TITLE
Check for failure from curl request before parsing monitored resource.

### PR DESCRIPTION
**Pull request by bmoyles0117**

### CODE REVIEW

File: src/write_gcm.c [hunk 1]
```diff
@@ -2210,12 +2210,18 @@ static monitored_resource_t *wg_monitored_resource_create_from_metadata_agent(
...
-  wg_curl_get_or_post(&response_buffer, query, body, headers, num_headers, 0);
-  monitored_resource_t *resource = parse_monitored_resource(response_buffer,
-      project_id);
-   sfree(response_buffer);
-   return resource;
+  char *response = NULL;
+  if (wg_curl_get_or_post(&response, query, body, headers, num_headers, 0)
+      != 0) {
+    ERROR("write_gcm: wg_monitored_resource_create_from_metadata_agent: "
+          "Failed to get monitored resource.");
```
Comment (dhrupadb) [inline src/write_gcm.c]: Let's change the comment to something on the lines of "Failed to get monitored resource from local metadata agent service".

**Comment by bmoyles0117** [comment]
Friendly ping @igorpeshansky, would love to get this merged.

**Comment by dhrupadb** [comment]
Merged for now as this will be incorporated in PR #98

File: src/write_gcm.c [hunk 2]
```diff
@@ -2210,12 +2210,19 @@ static monitored_resource_t *wg_monitored_resource_create_from_metadata_agent(
...
-      project_id);
-   sfree(response_buffer);
-   return resource;
+  char *response = NULL;
+  if (wg_curl_get_or_post(&response, query, body, headers, num_headers, 0)
+      != 0) {
+    ERROR("write_gcm: wg_monitored_resource_create_from_metadata_agent: "
+          "Failed to get monitored resource from local metadata agent "
+          "service.");
+    return NULL;
```
Comment (igorpeshansky) [inline src/write_gcm.c]: `goto error;`

**Pull request merged**